src/classes/RD2_ElevateIntegrationService_TEST.cls

```diff
@@ -0,0 +1,141 @@
+/*
+    Copyright (c) 2020, Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020
+* @group Recurring Donations
+* @description Unit Tests for the Elevate Integration Service
+*/
+@IsTest
+private class RD2_ElevateIntegrationService_TEST {
+    private static final TEST_SObjectGateway.RecurringDonationGateway rdGateway = new TEST_SObjectGateway.RecurringDonationGateway();
+    private static final TEST_SObjectGateway.OpportunityGateway oppGateway = new TEST_SObjectGateway.OpportunityGateway();
+
+    private static final String COMMITMENT_ID = '0009989378376210918302112371207242332342432';
+
+    /****
+    * @description Creates data required for unit tests
+    */
+    @TestSetup
+    private static void setUp() {
+        Contact contact = UTIL_UnitTestData_TEST.getContact();
+        insert contact;
+    }
+
+
+    /***
+    * @description Verifies that any existing Opps are linked to a new RD if the CommitmentId matches
+    */
+    @isTest
+    private static void shouldLinkOppsToRecurringDonation() {
```

Comment:
This test should be split into various tests where each test name specifies rule when Opp is matched with a Recurring Donation or when it is omitted from matching (one test per concept).

Reply:
This actually is a single concept - verifying that the right Opp is updated _while_ the other Opps are not updated. I could split these out, but then it's not testing that only the one Opp is updated which is the goal here. I could also create another test method that re-tests that a single Opp is updated, but that seems redundant since this method is testing that concept.

Reply:
ok, I see, so this should be `shouldLinkOppsToRDMatchingCommitmentId() `?
Can we also verify AC3 that if the Commitment ID is matching existing Opp but the Opp is already assigned to an RD, the Opp will not be matched with the just created RD? For example: `shouldNotLinkAlreadyAssignedOppsToRDMatchingCommitmentId() `?